src/main/catalog-pusher.ts

```diff
@@ -0,0 +1,26 @@
+import { autorun, toJS } from "mobx";
+import { broadcastMessage, subscribeToBroadcast } from "../common/ipc";
+import { CatalogEntityRegistry} from "../common/catalog-entity-registry";
+import "../common/catalog-entities/kubernetes-cluster";
+
+export class CatalogPusher {
+  static init(catalog: CatalogEntityRegistry) {
+    new CatalogPusher(catalog).init();
+  }
+
+  constructor(private catalog: CatalogEntityRegistry) {}
```

Comment:
Should be a `private` then if static `init()` is entry point?
Also `init()` probably should return disposer from `autorun` to make clean up at some point?

Reply:
Improved.